edb/schema/constraints.py

```diff
@@ -733,6 +736,12 @@ def _populate_concrete_constraint_attrs(
                 f'subjectexpr is already defined for {name!r}'
             )
 
+        if (isinstance(subject_obj, s_scalars.ScalarType)
+                and constr_base.get_is_aggregate(schema)):
+            raise errors.InvalidConstraintDefinitionError(
+                f'Constraint {name} may not be used on scalar types'
```

Comment:
```suggestion
                f'{constr_base.get_verbosename(schema)} may not be used on scalar types'
```

Reply:
Hm. Sorry, `abstract constraint 'std::exclusive'` is a tad misleading and irrelevant in this context, so let's do
`f'constraint {constr_base.get_displayname(schema)} may not ...` (to elide the word "abstract")